File: Java/Easy/Accepted/number-of-different-integers-in-a-string_5.java
class Solution {
    public int numDifferentIntegers(String word) {
        
        int n = word.length();
        Set<String> set = new HashSet<>();
        for (int i = 0; i < n; i++) {
            if (Character.isDigit(word.charAt(i))) {
                StringBuilder sb = new StringBuilder();
                while (i < n && Character.isDigit(word.charAt(i))) {
                    sb.append(word.charAt(i));
                    i++;
                }
                while (sb.length() > 1 && sb.charAt(0) == '0') {
                    sb.deleteCharAt(0);
                }
                set.add(sb.toString());
            }
        }
        return set.size();
    }
}

File: C++/Easy/Accepted/number-of-different-integers-in-a-string_5.cpp
class Solution {
public:
    int numDifferentIntegers(string word) {
        
        int n = word.size();
        unordered_set<string> s;
        string t;
        for(int i = 0; i < n; i++){
            if(word[i] >= '0' && word[i] <= '9'){
                t += word[i];
                while(i + 1 < n && word[i + 1] >= '0' && word[i + 1] <= '9'){
                    t += word[i + 1];
                    i++;
                }
                while(t.size() > 1 && t[0] == '0'){
                    t = t.substr(1);
                }
                s.insert(t);
                t = "";
            }
        }
        return s.size();
    }
};